[WIP] Broadcasting profiles

Palakis opened the pull request:
**This is currently a Work in Progress. Please do NOT merge.**

This feature allows management of several sets of Live Broadcasting settings called "profiles" or "broadcasting profiles". A user can select which profile (set of server credentials, settings and encoding settings) to use when broadcasting using Mixxx.

This feature is phase 1 (out of 2) of my GSoC project for Mixxx, so the UI implemented during this phase paves the way for phase 2: multiple simultaneous broadcasting connections (a.k.a Multi-broadcasting)

More details on the project here: https://www.mixxx.org/wiki/doku.php/multi-broadcasting

__Tasks:__
- Implement the BroadcastProfile class:
  - [X] Basic structure
  - [X] XML read/write code
  - [x] Validate proper functioning
  - [ ] Write unit tests
- Refactor the BroadcastSettings class
  - [x] QList of BroadcastProfile instances
  - [x] List profiles from filesystem
  - [x] Get/set `CurrentProfile` key in mixxx.cfg referencing the name of the currently selected profile
  - [ ] Make sure only one instance of BroadcastSettings exists
- [ ] Refactor UI of Live Broadcasting preferences panel (DlgPrefBroadcast)

Commit:
preferences: add WIP class for broadcasting profiles

Commit:
xml profile schema: refactor tag names to SnakeCase

Commit:
Fix errors in BroadcastProfile and make sure it builds

I initially forgot to add broadcastprofile.cpp into the list
of source files in build/depends.py, so I didn't noticed all the
errors and mistakes I've made while writing this earlier.
This commit aims to fix these errors and have the BroadcastProfile
class to build.

daschuer commented on src/preferences/broadcastprofile.cpp:
```diff
@@ -0,0 +1,459 @@
+
+BroadcastProfile::BroadcastProfile(const QString& profileName) {
+    // TODO(Palakis, June 2nd 2017):
+    //   Strip forbidden characters from the profile name
```
there must be a qt class that escapes user strings

Commit:
(WIP) BroadcastSettings and BroadcastProfile refactor

No idea if this works. I just want to make sure it is saved
somewhere out of my computer.
I may squash this with other commits later.

Commit:
BroadcastSettings: fix behaviour of filename gen. method

Commit:
BroadcastProfile: fix XML load/save code

Both ways didn't handled the document's root tag.
Now, profile values are saved properly.

Commit:
BroadcastSettings: enhance default profile creation

Commit:
BroadcastSettings: add qDebug statements

Commit:
BroadcastProfile: remove extra qualifications in header file

Commit:
Merge branch 'master' of https://github.com/mixxxdj/mixxx into broadcast-profiles

Commit:
BroadcastProfile: fix XML file load code

Commit:
(WIP) Preferences: add profiles to Live Broadcasting UI

Commit:
Live Broadcasting prefs: use QListView instead

Commit:
(WIP) Broadcasting prefs: implement logic behind new UI

Palakis commented:
Work for phase 1 has been slimmed down to only the XML profile subsystem, as tasks of this PR didn't focused enough on writing a good foundation for it.  
This branch has been forked at https://github.com/mixxxdj/mixxx/pull/1283/commits/b2d089d9d6158c1756068501b2e122ae4fc3a906 to a new branch for PR #1283.
